chain/dependencies.go

```diff
@@ -287,3 +287,12 @@ type AuthFactory interface {
 	MaxUnits() (bandwidth uint64, compute uint64)
 	Address() codec.Address
 }
+
+type Registry struct {
+	Action ActionRegistry
+	Auth   AuthRegistry
+	Output OutputRegistry
+}
+
+// RegistryFactory is the factory function, provided to the VM initializer that provides the registries for the actions, auth and output.
+type RegistryFactory func() Registry
```

Comment:
Can we change from an interface returning a struct of all of these types to an interface that supports `ActionRegistry()`, `AuthRegistry()`, and `OutputRegistry()` individually?

We can have a struct that implements that interface and takes the exact same fields, that way we can pass it around more easily and call a single function on the interface to get the value we care about.

This will also make it more ergonomic when we have a type that needs only the one of those values. The callee can then declare its own dependency and the caller can pass in the existing interface.

Reply:
done.